cloudinit/net/network_manager.py

```diff
@@ -0,0 +1,369 @@
+        if not ipv4_mtu == device_mtu:
+            LOG.warning(
+                "Network config: ignoring %s device-level mtu:%s"
+                " because ipv4 subnet-level mtu:%s provided.",
+                iface["name"],
+                device_mtu,
+                ipv4_mtu,
+            )
+
+        # Parse type-specific properties
+        for nm_prop, key in _prop_map[if_type].items():
+            if key not in iface:
+                continue
+            if iface[key] is None:
+                continue
+            if isinstance(iface[key], bool):
+                self.config[if_type][nm_prop] = (
+                    "true" if iface[key] else "false"
+                )
+            else:
+                self.config[if_type][nm_prop] = str(iface[key])
+
+        # These ones need special treatment
+        if if_type == "ethernet":
+            if iface["wakeonlan"] is True:
+                # NM_SETTING_WIRED_WAKE_ON_LAN_MAGIC
+                self.config["ethernet"]["wake-on-lan"] = str(0x40)
+            if ipv4_mtu is not None:
+                self.config["ethernet"]["mtu"] = str(ipv4_mtu)
+            if iface["mac_address"] is not None:
+                self.config["ethernet"]["mac-address"] = self.mac_addr(
+                    iface["mac_address"]
+                )
+        if if_type == "vlan" and "vlan-raw-device" in iface:
+            self.config["vlan"]["parent"] = renderer.con_ref(
+                iface["vlan-raw-device"]
+            )
+        if if_type == "bridge":
+            # Bridge is ass-backwards compared to bond
+            for port in iface["bridge_ports"]:
+                port = renderer.get_conn(port)
+                port._set_default("connection", "slave-type", "bridge")
+                port._set_default("connection", "master", self.con_uuid())
+            if iface["mac_address"] is not None:
+                self.config["bridge"]["mac-address"] = self.mac_addr(
+                    iface["mac_address"]
+                )
+        if if_type == "infiniband" and ipv4_mtu is not None:
+            self.config["infiniband"]["transport-mode"] = "datagram"
+            self.config["infiniband"]["mtu"] = str(ipv4_mtu)
+            if iface["mac_address"] is not None:
+                self.config["infiniband"]["mac-address"] = self.mac_addr(
+                    iface["mac_address"]
+                )
+
+        # Finish up
+        if if_type == "bridge" or not self.config.has_option(
+            if_type, "mac-address"
+        ):
+            self.config["connection"]["interface-name"] = iface["name"]
+
+    def __str__(self):
+        """
+        Stringify.
+        """
+
+        buf = io.StringIO()
+        self.config.write(buf, space_around_delimiters=False)
+        header = "# Generated by cloud-init. Changes will be lost.\n\n"
+        return header + buf.getvalue()
+
+
+class Renderer(renderer.Renderer):
+    """Renders network information in a /etc/sysconfig format."""
```

Comment:
This description should be changed to something like:

```suggestion
    """Renders network information in the NetworkManager keyfile format."""
```

Reply:
Applied. Thanks!